backend/danswer/db/chat.py

```diff
@@ -81,17 +87,47 @@ def get_chat_sessions_by_slack_thread_id(
     return db_session.scalars(stmt).all()
 
 
+def get_first_messages_for_chat_sessions(
+    chat_session_ids: list[int], db_session: Session
+) -> dict[int, str]:
+    subquery = (
+        select(ChatMessage.chat_session_id, func.min(ChatMessage.id).label("min_id"))
+        .where(
+            and_(
+                ChatMessage.chat_session_id.in_(chat_session_ids),
+                ChatMessage.message_type == MessageType.USER,  # Select USER messages
+            )
+        )
+        .group_by(ChatMessage.chat_session_id)
+        .subquery()
+    )
+
+    query = select(ChatMessage.chat_session_id, ChatMessage.message).join(
+        subquery,
+        (ChatMessage.chat_session_id == subquery.c.chat_session_id)
+        & (ChatMessage.id == subquery.c.min_id),
+    )
+
+    first_messages = db_session.execute(query).all()
+    return dict([(row.chat_session_id, row.message) for row in first_messages])
+
+
 def get_chat_sessions_by_user(
     user_id: UUID | None,
     deleted: bool | None,
     db_session: Session,
     include_one_shot: bool = False,
+    only_one_shot: bool = False,
 ) -> list[ChatSession]:
     stmt = select(ChatSession).where(ChatSession.user_id == user_id)
 
-    if not include_one_shot:
+    if not include_one_shot and not only_one_shot:
         stmt = stmt.where(ChatSession.one_shot.is_(False))
 
+    if only_one_shot:
+        stmt = stmt.where(ChatSession.one_shot.is_(True))
+        stmt = stmt.order_by(desc(ChatSession.id))
```

Comment:
Why is this one ordered whereas the other use cases are not? If it makes sense to order then, we can just always do it, this should not increase latency in any non-trivial way.

Other thing is, technically IDs are strictly increasing but I feel it's better to use time_created instead of ID